crawler-cli/src/main/java/org/gbif/crawler/pipelines/dwca/DwCAToAvroCallBack.java

```diff
@@ -0,0 +1,83 @@
+package org.gbif.crawler.pipelines.dwca;
+
+import org.gbif.common.messaging.AbstractMessageCallback;
+import org.gbif.common.messaging.api.messages.DwcaValidationFinishedMessage;
+import org.gbif.pipelines.core.io.DwCAReader;
+import org.gbif.pipelines.io.avro.ExtendedRecord;
+
+import java.io.IOException;
+import java.io.OutputStream;
+import java.net.URI;
+import java.util.Objects;
+
+import org.apache.avro.file.DataFileWriter;
+import org.apache.avro.io.DatumWriter;
+import org.apache.avro.specific.SpecificDatumWriter;
+import org.apache.hadoop.conf.Configuration;
+import org.apache.hadoop.fs.FileSystem;
+import org.apache.hadoop.fs.Path;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * Call back which is called when the {@link org.gbif.common.messaging.api.messages.DwcaValidationFinishedMessage } is received.
+ */
+public class DwCAToAvroCallBack extends AbstractMessageCallback<DwcaValidationFinishedMessage> {
+
+  private static final Logger LOG = LoggerFactory.getLogger(DwCAToAvroCallBack.class);
+  private final DwCAToAvroConfiguration configuration;
+
+  DwCAToAvroCallBack(DwCAToAvroConfiguration configuration) {
+    Objects.requireNonNull(configuration, "Configuration cannot be null");
+    this.configuration = configuration;
+  }
+
+  @Override
+  public void handleMessage(DwcaValidationFinishedMessage dwcaValidationFinishedMessage) {
+    LOG.info("Received Download finished validation message {}", dwcaValidationFinishedMessage);
+    LOG.info("Verifying the configuration parameters for dataset {}, before exporting",
+             dwcaValidationFinishedMessage.getDatasetUuid());
+    DwCAToAvroPaths paths = DwCAToAvroPaths.from(configuration, dwcaValidationFinishedMessage);
+
+    DatumWriter<ExtendedRecord> writer = new SpecificDatumWriter<>();
```

Comment:
writer can be moved inside the try: DataFileWriter<ExtendedRecord> dataFileWriter = new DataFileWriter<>(new SpecificDatumWriter<ExtendedRecord>())
           .create(ExtendedRecord.getClassSchema(), extendedRepoPath)) ??